src/app/upload/upload.component.ts

```diff
@@ -85,6 +85,12 @@ export class UploadComponent implements OnInit, DoCheck {
         this.changedClass = !this.versionUndeletedHaveChanged();
       } else if (this.version.changed(null, false)) {
         this.changedClass = true; // version itself changed, not files
+      } else if (this.version.nonMatchingFiles()) {
```

Comment:
This is what I feel worst about.  For DRAFT stories, I wanted to let the user save any audio files, but then show the mismatch-error after processing.  Without this special case, the error wouldn't show up in Version-uploader box (presumably because the version hasn't `changed()`) - but would show up in the Story-status-box, and prevent publishing.